sktime/utils/validation/forecasting.py

```diff
@@ -101,22 +100,32 @@ def check_X(
     )
 
 
-def check_y(y, allow_empty=False, allow_constant=True, enforce_index_type=None):
+def check_y(
```

Comment:
Could we still revert the changes here (removing the option to enforce_univariate) as discussed in the previous PR?

Reply:
@mloning and @fkiraly I'm using the scikit-learn checks for the metric input validations, which allow multivariate inputs. However, the scikit-learn checks won't work for validating the input `y_train` info for those metrics that need it (MASE and its related metrics). 

One option would be to keep the enforce_univariate option (which is also useful down the line for multivariate forecasting problems). 

Other option is that I create some separate validation that just applies to `y_train` metric inputs.

Reply:
@mloning should be backed out now